### PR TITLE
The `design-hs` library. 

### PR DESCRIPTION
**Pull request by asheshambasta**
This will be a very large PR with quite a few changes summed into one. 

The goal of this PR is to conclude the initial work on the design-hs lib. & continue on to some actual prototyping.

### CODE REVIEW

File: exe/bin/Conf/Parse.hs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ where
+  helpText =
+    name
+      <> " directory under which to construct the tree, existing files will be overwritten."
+  long' = name <> "-dir"
+
+confParserInfo :: ParserInfo Conf
+confParserInfo = info
+  (confParser <**> helper)
+  (fullDesc <> progDesc "design-hs: SmartCoop design examples" <> header
```
Comment (thusc) [inline exe/bin/Conf/Parse.hs:30]: I guess this naming is taken from an older version of the design.smart.coop site, but we replaced SmartCoop by simply Smart, and also, make sure to call it Smart design system, not just Smart design.
Maybe to be specific, we should say it's a "Smart design system implementation" of "An implementation of the Smart design system in Haskell".
(Same for the line below)

File: exe/bin/Conf/Types.hs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module Conf.Types
+  ( FilesystemConf(..)
+  , Conf(..)
+  ) where
+
+-- | Run configuration
```
Comment (thusc) [inline exe/bin/Conf/Types.hs:6]: Add something like "See Conf/Parse.hs" to see how `FilesystemConf` can be created ?

File: nix/contents.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+in { nixpkgs ? defNixpkgs }:
+
+let inherit (nixpkgs.lib.attrsets) getAttrFromPath;
+in {
+  # Lists all packages made available through this nix project.
+  # The format is `{ <pkgName> : <pkgDir> }` (we refer to this as pInfo)
+  # The used directory should be the path of the directory relative to the root of the project.
+  pkgList = { design-hs-lib = ../lib;
+              design-hs-exe = ../exe;
```
Comment (thusc) [inline nix/contents.nix:14]: Regarding a previous comment: would it make sense to have `design-hs-html` appear here ?

**Comment by thusc** [comment]
@asheshambasta I just tried to render the static site locally. Here is the link that appears in the index for the accordion examples: `/nix/store/xra2x1fd2dk2n3flfwkmji81d67ifs22-design-hs-html/html/examples/accordions.html`. This should be `/examples/accordions.html`.

**Comment by asheshambasta** [comment]
> @asheshambasta I just tried to render the static site locally. Here is the link that appears in the index for the accordion examples: `/nix/store/xra2x1fd2dk2n3flfwkmji81d67ifs22-design-hs-html/html/examples/accordions.html`. This should be `/examples/accordions.html`.

@thusc Correct, I've identified and and it should be fixed now.

File: nix/contents.nix (hunk 3 above)
Comment (asheshambasta) [reply nix/contents.nix:14]: Technically, I'd say the "html" is the _output_ of the `design-hs-exe`; and not a package in itself. The `design-hs-{lib, exe}` packages just provide you with: 
1. A library to generate HTML 
2. An executable to generate some example HTML & documentation. 
I personally don't think it should be considered a package per se. 

However, I do agree that we should be able to build HTML using `nix-build -A design-hs-html`.

File: exe/bin/Conf/Parse.hs (hunk 1 above)
Comment (asheshambasta) [reply exe/bin/Conf/Parse.hs]: @thusc This should be fixed, let me know if its correct now.

File: nix/contents.nix (hunk 3 above)
Comment (asheshambasta) [reply nix/contents.nix:14]: @thusc this has been changed and I've committed my changes.

**Comment by thusc** [comment]
Very nice, thanks a lot :-)

**Pull request merged**